mm2src/coins/lightning.rs

```diff
@@ -2,68 +2,106 @@
 use super::{lp_coinfind_or_err, MmCoinEnum};
 #[cfg(not(target_arch = "wasm32"))]
 use crate::utxo::rpc_clients::UtxoRpcClientEnum;
+use crate::utxo::utxo_common::big_decimal_from_sat_unsigned;
 #[cfg(not(target_arch = "wasm32"))]
 use crate::utxo::utxo_common::UtxoTxBuilder;
 use crate::utxo::BlockchainNetwork;
 #[cfg(not(target_arch = "wasm32"))]
 use crate::utxo::{sat_from_big_decimal, FeePolicy, UtxoCommonOps, UtxoTxGenerationOps};
 use crate::{BalanceFut, CoinBalance, FeeApproxStage, FoundSwapTxSpend, HistorySyncState, MarketCoinOps, MmCoin,
             NegotiateSwapContractAddrErr, SwapOps, TradeFee, TradePreimageFut, TradePreimageValue, TransactionEnum,
-            TransactionFut, UtxoStandardCoin, ValidateAddressResult, WithdrawFut, WithdrawRequest};
+            TransactionFut, UtxoStandardCoin, ValidateAddressResult, WithdrawError, WithdrawFut, WithdrawRequest};
 use async_trait::async_trait;
 use bigdecimal::BigDecimal;
 use bitcoin::blockdata::script::Script;
 use bitcoin::hash_types::Txid;
+#[cfg(not(target_arch = "wasm32"))] use bitcoin::hashes::Hash;
```

Comment:
> If think we can use the cfg_native macro like you did [here](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/lightning/ln_utils.rs#L14).

Done

> Is it possible to use lightning channels in WASM?

I don't think anything prevents that. The main crate in `rust-lightning` already compiles to wasm and has a lot of code relating to channels. A lightning channel, in the end, is a 2-of-2 Multisig transaction. The additional helper crates are the reason WASM compilation is not working. `rust-lightning` is designed so that we can use our own code instead of these additional crates if any other functionalities are needed, that's why I already ported `lightning-background-processor` and `lightning-persister` to mm2 and when it's time to work on WASM, I will modify them accordingly. I will also need to port `lightning-net-tokio` then.

> I guess some lightning crates don't support WASM, so what do you think if we just turn off the lightning coin until we port these crates into WASM?

I don't think this is a good idea since I use dispatcher_v2 for enabling the lightning coin.

Reply:
> I don't think this is a good idea since I use dispatcher_v2 for enabling the lightning coin.

This can just allow us to avoid duplicate functions like
```rust
#[cfg(target_arch = "wasm32")]
fn foo() {}

#[cfg(not(target_arch = "wasm32"))]
fn foo() {}
```